llvm/lib/Transforms/InstCombine/InstCombineAndOrXor.cpp

```diff
@@ -3421,6 +3421,42 @@ Value *InstCombinerImpl::foldAndOrOfICmps(ICmpInst *LHS, ICmpInst *RHS,
   return foldAndOrOfICmpsUsingRanges(LHS, RHS, IsAnd);
 }
 
+Value *foldAorBZero(BinaryOperator &I, InstCombiner::BuilderTy &Builder) {
+  Value *Op0 = I.getOperand(0);
+  Value *Op1 = I.getOperand(1);
+  if (!Op0->hasOneUse() || !Op1->hasOneUse())
+    return nullptr;
+
+  Value *Cmp1, *Cmp2, *Cmp3, *Cmp4;
+  bool IsOp0 = match(Op0, m_And(m_Value(Cmp1), m_Value(Cmp2)));
+  bool IsOp1 = match(Op1, m_And(m_Value(Cmp3), m_Value(Cmp4)));
+  if (!IsOp0 || !IsOp1)
+    return nullptr;
+
+  Value *A;
+  Value *B;
```

Comment:
Declare variables where you initialize them.

Reply:
i guess those aren't needed now